doc/source/admin/dependency_resolvers.rst

```diff
@@ -5,56 +5,68 @@ Dependency Resolvers in Galaxy
-``dependency_resolvers_conf.xml`` configuration file in the Galaxy ``config/`` directory.
+specifies a mapping between the Galaxy web user interface and the tool command line, and (2) the actual command-line
+tools, known as Galaxy *tool dependencies*, which must be installed and available on the system(s) where Galaxy is
+configured to run those tools. The job script that Galaxy uses to run a job includes commands (such as changes to the
+``PATH`` environment variable) that are generated by *dependency resolvers*. These same dependency resolvers are used by
+the Galaxy administrative UI to display whether an installed tool's dependencies have been installed on the Galaxy
+server, and to show how they will be resolved at job runtime.  There is a default dependency resolver configuration but
+administrators can provide their own configuration using the ``dependency_resolvers`` configuration option in Galaxy's
+configuration file, ``galaxy.yml``. Previously this configuration was stored in a separate XML file,
+``dependency_resolvers_conf.xml``. Loading the dependency resolvers configuration from that XML file is deprecated but
+still supported, however, the documentation and sample configuration file for the XML format can only be found in Galaxy
+releases prior to 21.09.
 
-The binding between tool XML and the system tools they need to run is specified in the tool XML using ``<requirement>``
-tags, for example
+.. note::
+
+    The *tool XML* referred to below is different from the deprecated *dependency resolvers XML* referred to above.
+
+The binding between tool XML and the command-line tools they need to run is specified in the tool XML using
+``<requirement>`` tags, for example:
 
 .. code-block:: xml
 
     <requirement type="package" version="0.7.10.039ea20639">bwa</requirement>
 
-In some cases these requirement tags can be specified without a version
+In some cases these requirement tags can be specified without a version:
 
 .. code-block:: xml
 
     <requirement type="package">bedtools</requirement>
 
-These declared requirements are passed as inputs to the dependency resolver.
+These declared requirements are passed as inputs to the dependency resolver in order to generate the environmental setup
+in the job script so that the correct tool dependencies required by the tool are found on the ``$PATH``.
 
 Default Dependency Resolvers
 ----------------------------
 
-The default configuration of dependency resolvers is equivalent to the following ``dependency_resolvers_conf.xml``
+The default configuration of dependency resolvers is equivalent to the following configuration in ``galaxy.yml``:
 
-.. code-block:: xml
+.. code-block:: yaml
 
-  <dependency_resolvers>
-    <tool_shed_packages />
-    <galaxy_packages />
-    <conda />
-    <galaxy_packages versionless="true" />
-    <conda versionless="true" />
-  </dependency_resolvers>
+    galaxy:
+      dependency_resolvers:
+        - type: tool_shed_packages
```

Comment:
@natefoo Unless I am misunderstanding, I suspect `type` may not work. Since this block is meant to be (optionally) specified inline , it will be then processed as part of `galaxy.yml`, and that code interprets the `type` key as the expected datatype of the config option: https://github.com/galaxyproject/galaxy/blob/c1eae49615665aa01840e47e5c917d4496a34795/lib/galaxy/config/__init__.py#L310

Even if we did not have that method (or added a special case), I think it would be semantically incorrect (our config system, at least now, does not allow config options to have their own scope for keys (i.e., `type` will mean the same for all keys regardless of parent node or nesting level).

Does this make sense, or am I missing something?

Reply:
You're right - it's fine. The schema `type` key is, of course, not the same as a `type` key in the config! And we do not specify in the schema what keys are allowed for the sequence type of this option, just their type (`any`).  So all is fine, false alarm, sorry!